tests/BasicRegistryTests.cpp

```diff
@@ -917,7 +1215,371 @@ TEST_CASE("BasicRegistryTests::wstrings", "[registry][get_registry_string]")
             REQUIRE(e.GetErrorCode() == HRESULT_FROM_WIN32(ERROR_UNSUPPORTED_TYPE));
         }
     }
+
+    SECTION("set_value_nothrow/get_value_string_nothrow: with open key")
+    {
+        wil::unique_hkey hkey;
+        REQUIRE_SUCCEEDED(wil::reg::create_key_nothrow(HKEY_CURRENT_USER, testSubkey, &hkey, wil::reg::key_access::readwrite));
+
+        for (const auto& value : stringTestArray)
+        {
+            WCHAR result[100]{};
+            REQUIRE_SUCCEEDED(wil::reg::set_value_nothrow(hkey.get(), stringValueName, value.c_str()));
+            REQUIRE_SUCCEEDED(wil::reg::get_value_string_nothrow(hkey.get(), stringValueName, result));
+            REQUIRE(result == value);
+
+            // and verify default value name
+            REQUIRE_SUCCEEDED(wil::reg::set_value_nothrow(hkey.get(), nullptr, value.c_str()));
+            REQUIRE_SUCCEEDED(wil::reg::get_value_string_nothrow(hkey.get(), nullptr, result));
+            REQUIRE(result == value);
+        }
+
+        WCHAR too_small_result[4]{};
+        // fail get* if the buffer is too small
+        REQUIRE_SUCCEEDED(wil::reg::set_value_string_nothrow(hkey.get(), stringValueName, L"Test"));
+        DWORD expectedSize{};
+        auto hr = wil::reg::get_value_string_nothrow(hkey.get(), stringValueName, too_small_result, &expectedSize);
+        REQUIRE(hr == HRESULT_FROM_WIN32(ERROR_MORE_DATA));
+        REQUIRE(expectedSize == 12);
+        WCHAR valid_buffer_result[5]{};
+        REQUIRE_SUCCEEDED(wil::reg::get_value_string_nothrow(hkey.get(), stringValueName, valid_buffer_result, &expectedSize));
+        REQUIRE(expectedSize == 10);
+        REQUIRE(0 == wcscmp(valid_buffer_result, L"Test"));
+
+        // fail get* if the value doesn't exist
+        hr = wil::reg::get_value_string_nothrow(hkey.get(), (std::wstring(stringValueName) + L"_not_valid").c_str(), too_small_result, &expectedSize);
+        REQUIRE(hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+        REQUIRE(expectedSize == 0);
+
+        // fail if get* requests the wrong type
+        try
+        {
+            wil::reg::try_get_value_dword(hkey.get(), stringValueName);
+            // should throw
+            REQUIRE_FALSE(true);
+        }
+        catch (const wil::ResultException& e)
+        {
+            REQUIRE(e.GetErrorCode() == HRESULT_FROM_WIN32(ERROR_UNSUPPORTED_TYPE));
+        }
+    }
+    SECTION("set_value_nothrow/get_value_string_nothrow: with string key")

```

Comment:
I think some of these tests are still gated on exceptions-enabled. Should we extract them?